ext/standard/streamsfuncs.c

```diff
@@ -673,8 +673,9 @@ static int stream_array_from_fd_set(zval *stream_array, fd_set *fds)
 				} else {
 					dest_elem = zend_hash_update(ht, key, elem);
 				}
-
-				zval_add_ref(dest_elem);
+				if (dest_elem) {
```

Comment:
This check shouldn't be necessary, as zend_hash_update is infallible.

Reply:
Copied from the original code...
Resolved.